internal/server/grants.go

```diff
@@ -61,29 +59,25 @@ func (a *API) CreateGrant(c *gin.Context, r *api.CreateGrantRequest) (*api.Creat
 		Privilege: r.Privilege,
 	}
 
-	err := access.CreateGrant(c, grant)
-	var ucerr data.UniqueConstraintError
-
-	if errors.As(err, &ucerr) {
-		grants, err := access.ListGrants(c, grant.Subject, grant.Resource, grant.Privilege, false, &models.Pagination{})
-
-		if err != nil {
-			return nil, err
-		}
-
-		if len(grants) == 0 {
-			return nil, fmt.Errorf("duplicate grant exists, but cannot be found")
-		}
+	// check if grant already exists
+	grants, err := access.ListGrants(c, grant.Subject, grant.Resource, grant.Privilege, false, &models.Pagination{})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I suspect  this should be in `access.CreateGrant` so that it's after the authorization